lib/brain.js

```diff
@@ -3074,35 +3075,48 @@ Brain.prototype.saveTx = function saveTx (tx) {
 }
 
 Brain.prototype.postTx = function postTx (tx) {
-  const postTxF = timedout => {
-    const updatedTx = _.assign(tx, { timedout })
+  const onFailedAttempt = err =>
+    console.error(
+      err.retriesLeft > 0 ?
+        "Error posting TX, will retry:" :
+        "Error posting TX:",
+      err.message
+    )
 
-    return this.trader.postTx(updatedTx)
-      .then(serverTx => ({ tx: serverTx }))
+  let timedout = false
+  const started = Date.now()
+  const post = () => {
+    timedout = timedout || Date.now() - started >= NETWORK_TIMEOUT_INTERVAL
+    return this.trader.postTx(_.set('timedout', timedout, tx))
+      .catch(err => {
+        if (err instanceof E.StaleError || err instanceof E.RatchetError) {
+          console.log(
+            err.name === 'StaleError' ?
+              "Local transaction is outdated, won't retry:" :
+              "Local transaction conflicts with server, won't retry:",
+            err.message
+          )
+          return Promise.reject(new pRetry.AbortError(err))
+        }
+        return Promise.reject(err)
```

Comment:
Same here

Reply:
Same here xP

the `pRetry.AbortError` tells `pRetry` not to retry the call in this case.